### ISSUE TITLE
`methods[key] is not a function` in `4.23.0`

### ISSUE THREAD

**Issue by Enngage** [algolia/algoliasearch-client-javascript#1511]
Hello,
 
 We started receiving following error when upgrading from `4.22.1` to `4.23.0`. No change in code, just the bump up of the npm package.
 
 ```bash
 TypeError: methods[key] is not a function
    at c:\dev\kontent-ai-learn-fe\.next\server\chunks\node_modules_1a6303._.js:6521:33
    at Array.forEach (<anonymous>)
    at Module.addMethods (c:\dev\kontent-ai-learn-fe\.next\server\chunks\node_modules_1a6303._.js:6519:26)
    at Module.createSearchClient (c:\dev\kontent-ai-learn-fe\.next\server\chunks\node_modules_1a6303._.js:7352:202)
    at Module.algoliasearch (c:\dev\kontent-ai-learn-fe\.next\server\chunks\node_modules_1a6303._.js:8912:25)
    at initAlgoliaSearchClient
```
    
Cheers!

**Comment by shortcuts** [comment]
cc @dhayab the methods listed here https://github.com/algolia/algoliasearch-client-javascript/pull/1509/files#diff-24b438c650124b5fa554b87c388a4381214d893bcd98dae3cf0fb87c7210fbd6R359-R365 should be in a createRecommendClient not at the root of the search client, because it won't be able to find them

weird that the CI did not caught that

**Comment by dhayab** [comment]
Thanks for your report @Enngage. We've just released **algoliasearch@4.23.1** which includes a fix for the issue you've been experiencing.

**Comment by shortcuts** [comment]
thanks for tackling this quickly @dhayab
